src/NServiceBus.Core/Transports/Learning/LearningTransportMessagePump.cs

```diff
@@ -260,11 +266,20 @@ await ProcessFile(transaction, messageId, messageProcessingCancellationToken)
                     log.Debug($"Failure while trying to complete receive transaction for  {filePath}({transaction.FileToProcess})" + filePath, ex);
                 }
 
+                try
+                {
+                    await onCompleted(new CompleteContext(messageId, wasAcknowledged, headers ?? new Dictionary<string, string>(), startedAt, DateTimeOffset.UtcNow, processingContext), messageProcessingCancellationToken).ConfigureAwait(false);
+                }
+                catch (Exception ex)
+                {
+                    log.Debug($"Failure when invoking OnComplete for {filePath}({transaction.FileToProcess})" + filePath, ex);
```

Comment:
Just continued to do what all other code does in this method but DEBUG feels off to me? perhaps WARN?

Reply:
I wonder if this shouldn't be handled in Core instead?

Reply:
Not sure, we do try catch the calls to both OnError and OnMessage?

Reply:
I guess it should be logged as a warning, and if we're happy with adding this to the list of Things That Every Transport Must Do then let's just leave it here.

Incidentally, it seems off that we are swallowing the exception when the transaction fails to complete and just logging a debug message, but that's outside the scope of this PR.

Reply:
now logging WARN